lib/src/modal_page/wolt_modal_sheet_page.dart

```diff
@@ -1,223 +1,49 @@
-  /// a the back button.
-  final Widget? leadingNavBarWidget;
-
-  /// A widget representing trailing widget in the navigation toolbar. This widget is usually
-  /// a the close button.
-  final Widget? trailingNavBarWidget;
-
-  /// Creates a page to be built within [WoltScrollableModalSheet].
-  const WoltModalSheetPage({
-    this.pageTitle,
-    this.navBarHeight,
-    this.sliverList,
-    this.singleChildContent,
-    this.topBarTitle,
-    this.heroImage,
-    this.heroImageHeight,
-    this.backgroundColor,
-    this.hasSabGradient,
-    this.enableDrag,
-    this.sabGradientColor,
-    this.forceMaxHeight = false,
-    this.scrollController,
-    this.stickyActionBar,
-    this.leadingNavBarWidget,
-    this.trailingNavBarWidget,
-    this.hasTopBarLayer,
-    this.isTopBarLayerAlwaysVisible,
-    this.topBar,
-  })  : assert(singleChildContent == null || sliverList == null,
-            "Either singleChildContent or sliverList must be provided"),
-        assert(!(topBar != null && hasTopBarLayer == false),
-            "When topBar is provided, hasTopBarLayer must not be false");
+  /// This is a shortcut for providing a list of Sliver widgets with only one Sliver widget.
+  final Widget child;
 
   /// Creates a [WoltModalSheetPage] with a single child main content.
-  factory WoltModalSheetPage.withSingleChild({
-    required Widget child,
-    Widget? pageTitle,
-    double? navBarHeight,
-    Widget? topBarTitle,
-    Widget? heroImage,
-    double? heroImageHeight,
-    Color? backgroundColor,
-    bool? hasSabGradient,
-    Color? sabGradientColor,
-    bool? enableDrag,
-    bool forceMaxHeight = false,
-    bool? isTopBarLayerAlwaysVisible,
-    bool? hasTopBarLayer,
-    ScrollController? scrollController,
-    Widget? stickyActionBar,
-    Widget? leadingNavBarWidget,
-    Widget? trailingNavBarWidget,
-    Widget? topBar,
-  }) {
-    return WoltModalSheetPage(
-      singleChildContent: child,
-      pageTitle: pageTitle,
-      navBarHeight: navBarHeight,
-      topBarTitle: topBarTitle,
-      heroImage: heroImage,
-      heroImageHeight: heroImageHeight,
-      backgroundColor: backgroundColor,
-      hasSabGradient: hasSabGradient,
-      enableDrag: enableDrag,
-      sabGradientColor: sabGradientColor,
-      forceMaxHeight: forceMaxHeight,
-      isTopBarLayerAlwaysVisible: isTopBarLayerAlwaysVisible,
-      hasTopBarLayer: hasTopBarLayer,
-      scrollController: scrollController,
-      stickyActionBar: stickyActionBar,
-      leadingNavBarWidget: leadingNavBarWidget,
-      trailingNavBarWidget: trailingNavBarWidget,
-      topBar: topBar,
-    );
-  }
-
-  /// Creates a [WoltModalSheetPage] with a custom [SliverList] main content.
-  factory WoltModalSheetPage.withCustomSliverList({
-    required Widget sliverList,
+  WoltModalSheetPage({
+    required this.child,
     Widget? pageTitle,
     double? navBarHeight,
     Widget? topBarTitle,
```

Comment:
```suggestion
    super.pageTitle,
    super.navBarHeight,
    super.topBarTitle,
```
Use `super` for these and follow up properties

```dart
  WoltModalSheetPage({
    required this.child,
    super.pageTitle,
    super.navBarHeight,
    super.topBarTitle,
    super.heroImage,
    super.heroImageHeight,
    super.backgroundColor,
    super.hasSabGradient,
    super.sabGradientColor,
    super.enableDrag,
    super.forceMaxHeight = false,
    super.isTopBarLayerAlwaysVisible,
    super.hasTopBarLayer,
    super.scrollController,
    super.stickyActionBar,
    super.leadingNavBarWidget,
    super.trailingNavBarWidget,
    super.topBar,
  }) : super(mainContentSlivers: [SliverToBoxAdapter(child: child)]);
```

Reply:
Update follow up properties for `super` too